### PR TITLE
add docs for fourier series

### PR DESCRIPTION
**Pull request by leosartaj**
I have also restructured docs for series in general.

### CODE REVIEW

**Comment by leosartaj** [comment]
Ping @flacjacket @jcrist

File: sympy/series/fourier.py [hunk 1]
```diff
@@ -365,8 +359,8 @@ def fourier_series(f, limits=None):
     computed fourier series rather than computing
     again.
 
-    eg. If Fourier series of x**2 is known
-    fourier series of x**2 - 1 can be found by shifting by -1
+    eg. If Fourier series of ``x**2`` is known
```
Comment (asmeurer) [inline sympy/series/fourier.py]: Correct grammar should be "e.g., if Fourier series..."

File: sympy/series/fourier.py [hunk 2]
```diff
@@ -365,8 +359,8 @@ def fourier_series(f, limits=None):
...
+    fourier series of ``x**2 - 1`` can be found by shifting by ``-1``.
```
Comment (asmeurer) [inline sympy/series/fourier.py]: "the Fourier series"

**Comment by asmeurer** [comment]
Add `:members:` to all the classes in the docs.

**Comment by leosartaj** [comment]
I made the changes. There is a problem with documentation of ``coeff_mul`` methods in sequences. The docstring of ``coeff_mul`` says to see the docstring of ``SeqBase.coeff_mul`` which is not helpful. Is there a way that they can display the docstring of ``SeqBase.coeff_mul``. Or other way can be to introduce private method? @asmeurer

File: doc/src/modules/series/fourier.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Fourier Series
+==============
+
+Provides methods to compute Fourier Series.
```
Comment (flacjacket) [inline doc/src/modules/series/fourier.rst]: "Fourier Series" to "Fourier series"

File: sympy/series/fourier.py [hunk 4]
```diff
@@ -38,11 +38,11 @@ def fourier_sin_seq(func, limits, n):
...
     """
-    limits should be of the form (x, start, stop)
-    x should be a symbol. Both start and stop should be bounded
+    limits should be of the form (x, start, stop).
+    x should be a symbol. Both start and stop should be bounded.
 
-    * If x is not given, x is determined from func
-    * if limits is None. limit of the form (x, -pi, pi) is returned
+    * If x is not given, x is determined from func.
+    * if limits is None. limit of the form (x, -pi, pi) is returned.
```
Comment (flacjacket) [inline sympy/series/fourier.py]: "if limits" to "If limits"

File: sympy/series/fourier.py [hunk 5]
```diff
@@ -154,8 +154,10 @@ def _eval_subs(self, old, new):
             return self
 
     def truncate(self, n=3):
-        """"returns the first n terms(non-zero) of the series
-        if n is none returns an iterator"""
+        """returns the first n terms(non-zero) of the series.
```
Comment (flacjacket) [inline sympy/series/fourier.py]: "returns" to "Returns"
"terms(non-zero)" to "(non-zero) terms"

File: sympy/series/fourier.py [hunk 6]
```diff
@@ -154,8 +154,10 @@ def _eval_subs(self, old, new):
...
+
+        if n is ``None`` returns an iterator.
```
Comment (flacjacket) [inline sympy/series/fourier.py]: "if" to "If"

File: sympy/series/fourier.py [hunk 7]
```diff
@@ -228,9 +226,7 @@ def shiftx(self, s):
         return self.func(sfunc, self.args[1], (self.a0, an, bn))
 
     def scale(self, s):
-        """
-        Scale function by a
-        term independent of x
+        """Scale function by a term independent of x.
```
Comment (flacjacket) [inline sympy/series/fourier.py]: "Scale the function"

**Comment by flacjacket** [comment]
I'm not sure a good way to deal with `coeff_mul` in the different classes, other than linking to the SeqBase method or making a new docstring for each of the methods

**Comment by leosartaj** [comment]
@flacjacket  I took care of you comments (Hopefully). I added ``SeqBase`` to the docs because it contains almost all the public methods. 
ping @jcrist

**Comment by jcrist** [comment]
Looks good to me. I think you're fine just referring to another docstring for the full documentation. @flacjacket : feel free to merge if you feel your comments have been addressed.

**Comment by flacjacket** [comment]
Yeah, looks good to me, thanks!

**Comment by leosartaj** [comment]
Thanks for reviewing this. @flacjacket @jcrist

**Pull request merged**